### PR TITLE
Feat api page

### PR DESCRIPTION
**Pull request by kieranslost**
This was wasnt coded with as much love as learntrack usually is so dont judge me tooo hard @deepspace1000 🙏 xD

### CODE REVIEW

File: frontend/src/pages/DetailPage.tsx [hunk 1]
```diff
@@ -1,15 +1,386 @@
...
+          result = currencyAmount * currenyRates[firstExchange];
+          let formattedResult = result.toFixed(2);
+          setCurrencyResult2(currencyAmount);
+          setCurrencyResult1(formattedResult);
+        }
+      }
+    }
+  }, [convertingCurrency]);
+
+  const getLatLong = async () => {
```
Comment (deepspace1000) [inline frontend/src/pages/DetailPage.tsx:100]: you could move these functions into there own file in `src/utils` so the code here gets easier to read.

File: frontend/src/pages/DetailPage.tsx [hunk 2]
```diff
@@ -1,15 +1,386 @@
...
+      let lat = response.data[0]?.lat;
+      let lon = response.data[0]?.lon;
+      setLat(lat);
+      setLon(lon);
+    } catch (error) {
+      console.error('Error:', error);
+    }
+  };
+
+  const getWeatherData = async () => {
```
Comment (deepspace1000) [inline frontend/src/pages/DetailPage.tsx:114]: This to

File: frontend/src/pages/DetailPage.tsx [hunk 3]
```diff
@@ -1,15 +1,386 @@
...
+    try {
+      const response = await axios.get(
+        `https://api.openweathermap.org/data/2.5/weather?lat=${getLat}&lon=${getLon}&units=metric&appid=${WEATHER_API_KEY}`,
+      );
+      setWeather(response.data);
+    } catch (error) {
+      console.error('Error:', error);
```
Comment (deepspace1000) [inline frontend/src/pages/DetailPage.tsx:121]: i guess we can leave that, even thought an error handling would be nicer, we don't have the time for it.

File: frontend/src/pages/DetailPage.tsx [hunk 4]
```diff
@@ -1,15 +1,386 @@
...
+      setLat(lat);
+      setLon(lon);
+    } catch (error) {
+      console.error('Error:', error);
+    }
+  };
+
+  const getWeatherData = async () => {
+    try {
+      const response = await axios.get(
```
Comment (deepspace1000) [inline frontend/src/pages/DetailPage.tsx:116]: If you can tell me the right answer to this question you won't have to implement it 🙃: 
how could you improve this? (Hint: there are at least 2 correct answers and 1 is enough (Hint: `use` Something that you already know and use)

File: frontend/src/pages/DetailPage.tsx [hunk 5]
```diff
@@ -1,15 +1,386 @@
...
+  const handleButtonClick = (name: string) => {
+    setExchangeName(name);
+    getCurrencyData(name);
+  };
+
   return (
     <>
-      <Grid container sx={{ flexGrow: 1, maxWidth: '100%' }}>
-        <Grid xs={12} sx={{ position: 'relative', padding: '3rem 5rem', flexWrap: 'nowrap' }}>
+      <Grid container sx={{ flexGrow: 1, maxWidth: '100%', justifyContent: 'center' }}>
```
Comment (deepspace1000) [inline frontend/src/pages/DetailPage.tsx:158]: You already know how much i love those grids 😂

File: frontend/src/pages/DetailPage.tsx [hunk 6]
```diff
@@ -1,15 +1,386 @@
...
+                  </tr>
+                  <tr>
+                    <td>Min Temp.:</td>
+                    <td style={{ textAlign: 'right' }}>{getMinTemp}°</td>
+                  </tr>
+                </tbody>
+              </table>
+            </WeatherInfoCard>
+          </WeatherCard>
+          <CurrencyExchangeCard>
```
Comment (deepspace1000) [inline frontend/src/pages/DetailPage.tsx:202]: You could move the whole `<CurrencyExchangeCard>` into a new component.

**Review by deepspace1000 — COMMENTED** [review]
I guess if you would do the same thing now it would already look a lot different 🙃. I just wrote some ideas what you could do. i would at least move the functions into there own file and the other comments are suggestions. just write a comment if you dont implement those.

File: frontend/public/env.js
[the remark this answers is not in the document]
Comment (kieranslost) [reply frontend/public/env.js:1]: I have no idea what this is and how it got here

**Review by deepspace1000 — APPROVED** [review]

**Pull request merged**